components/compliance-service/ingest/ingestic/ingestic.go

```diff
@@ -43,6 +43,26 @@ func (backend *ESClient) addDataToIndexWithID(ctx context.Context,
 	return err
 }
 
+// This method will support adding a document with a specified id
+func (backend *ESClient) upsertComplianceRunInfo(ctx context.Context, mapping mappings.Mapping, id string, runDateTime time.Time) error {
+	script := elastic.NewScript(fmt.Sprintf("ctx._source.last_run = %s", runDateTime))
+	update, err := backend.client.Update().
+		Index(mapping.Index).
+		Type(mapping.Type).
+		Id(id).
+		Script(script).
+		Upsert(map[string]interface{}{
+			"node_uuid": id,
+			"first_run": runDateTime,
+			"last_run":  runDateTime,
+		}).
+		Do(ctx)
+
+	logrus.Debugf("run info updated for node: %s", update.Id)
```

Comment:
This statement may fail if `update` is empty. And it is not guaranteed that the `Do` method always returns a non-empty response. In the error case, the method will return an empty response.

Reply:
thanks.. I'm actually taking out that update var as I as only using it for debug